### PR TITLE
F/decouple logwriter setup

### PR DESCRIPTION
**Pull request by lbudai**
forward port from 3.6

### CODE REVIEW

File: modules/afsocket/afsocket-dest.c [hunk 1]
```diff
@@ -353,9 +353,11 @@ afsocket_dd_setup_addresses_method(AFSocketDestDriver *self)
 }
 
 static void
-afsocket_dd_restore_connection(AFSocketDestDriver *self)
+afsocket_dd_restore_writer(AFSocketDestDriver *self)
 {
-  GlobalConfig *cfg = log_pipe_get_config(&self->super.super.super);
+  GlobalConfig *cfg;
```
Comment (lbudai) [inline modules/afsocket/afsocket-dest.c]: oh... f***k made a mistakde during conflict resolution, please do not merge until I fixed it :-)
(cfg is not initialized!!!)

File: modules/afmongodb/libmongo-client [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 394a33880dfcadc46bcfba8334209bcb55b42106
```
Comment (juhaszviktor) [inline modules/afmongodb/libmongo-client]: It can be good, but are you sure that you want to update libmongo-client in this PR?

Comment (lbudai) [inline modules/afmongodb/libmongo-client]: Hmmm... you are right I was in a hurry... I'll fix it.
Thanks Viktor!

**Comment by lbudai** [comment]
@juhaszviktor : updated

**Comment by juhaszviktor** [comment]
:+1:

**Pull request merged**